modules/blockbuilder/blockbuilder.go

```diff
@@ -471,6 +482,12 @@ func (b *BlockBuilder) getPartitionState(partition int32, commits kadm.OffsetRes
 }
 
 func (b *BlockBuilder) stopping(err error) error {
+	select {
+	case <-b.consumeStopped:
+	case <-time.After(2 * b.cfg.ConsumeCycleDuration):
```

Comment:
Maybe this is too long? Wasn't sure what a reasonable timeout is. I'd prefer for this to not be configurable, a good default should be good enough.

Reply:
We can set the terminationGracePeriod

https://cloud.google.com/blog/products/containers-kubernetes/kubernetes-best-practices-terminating-with-grace

Reply:
Good call. Extended the timeout and `TerminationGracePeriod` to 60s.